### DISCUSSION TITLE
scripts in subdirectories inside `.kenv/scripts`  ?

### DISCUSSION THREAD

**Discussion by CoffiDev** [johnlindquist/kit#294] [Q&A]
From my usage of kit I have the impression that only the files directly under `.kenv/scripts` directory are shown on the list of scripts you can run (on the app and using the `kit` command line).

Is there a way to configure kit to also "know about" scripts in subdirectories inside `.kenv/scripts`?

**Comment by johnlindquist** [comment]
Quick answer:

Yes. I'll make a demo later today of how you can build your own main menu (it's "just a script" too).

Longer answer:

I don't recommend it. `/scripts` and `/bin` stay in sync because scripts are conceptually "global" like when you add `bin` to your $PATH. So rather than subdirectories, I would find a naming convention for scripts I want to group together:

No:
`github/upload-file.js`
`github/create-gist.js`

Yes:
`github-upload-file.js`
`github-create-gist.js`

## multiple kenvs

Soon Script Kit will support creating multiple kenvs wherever you want. You can think of "personal", "company", "shared", etc types of environments or even a project environment. So each kenv would have its own scripts/bin dirs.

**Reply by CoffiDev** [reply]
Yeah, makes sense to me. I like the idea of thinking about scripts like adding a program to my global $PATH, also the naming convention is how I have been doing it and it has been working well.

I think the hype now is on the support for multiple kenvs! 

Thanks for the response :D
